demos/segmentation_demo/main.cpp

```diff
@@ -89,178 +64,141 @@ int main(int argc, char *argv[]) {
+                outHeight = outSizeVector[1];
+                outWidth = outSizeVector[2];
+                break;
+            case 4:
+                outChannels = outSizeVector[1];
+                outHeight = outSizeVector[2];
+                outWidth = outSizeVector[3];
+                break;
+            default:
+                throw std::runtime_error("Unexpected output blob shape. Only 4D and 3D output blobs are"
+                    "supported.");
         }
-        // -----------------------------------------------------------------------------------------------------
 
-        // --------------------------- 4. Loading model to the device ------------------------------------------
-        slog::info << "Loading model to the device" << slog::endl;
-        ExecutableNetwork executable_network = ie.LoadNetwork(network, FLAGS_d, configure(FLAGS_config));
-        // -----------------------------------------------------------------------------------------------------
-
-        // --------------------------- 5. Create infer request -------------------------------------------------
-        slog::info << "Create infer request" << slog::endl;
-        InferRequest infer_request = executable_network.CreateInferRequest();
-        // -----------------------------------------------------------------------------------------------------
-
-        // --------------------------- 6. Prepare input --------------------------------------------------------
-        /** Iterate over all the input blobs **/
-        /** Iterating over all input blobs **/
-        for (const auto & item : inputInfo) {
-            /** Creating input blob **/
-            Blob::Ptr input = infer_request.GetBlob(item.first);
-
-            /** Fill input tensor with images. First r channel, then g and b channels **/
-            size_t num_channels = input->getTensorDesc().getDims()[1];
-            size_t image_size = input->getTensorDesc().getDims()[3] * input->getTensorDesc().getDims()[2];
-
-            auto data = input->buffer().as<PrecisionTrait<Precision::U8>::value_type*>();
-
-            /** Iterate over all input images **/
-            for (size_t image_id = 0; image_id < imagesData.size(); ++image_id) {
-                /** Iterate over all pixel in image (r,g,b) **/
-                for (size_t pid = 0; pid < image_size; pid++) {
-                    /** Iterate over all channels **/
-                    for (size_t ch = 0; ch < num_channels; ++ch) {
-                        /**          [images stride + channels stride + pixel id ] all in bytes            **/
-                        data[image_id * image_size * num_channels + ch * image_size + pid] = imagesData.at(image_id).get()[pid*num_channels + ch];
-                    }
-                }
-            }
+        ExecutableNetwork executableNetwork = ie.LoadNetwork(network, FLAGS_d);
+        InferRequest inferRequest = executableNetwork.CreateInferRequest();
+
+        cv::VideoCapture cap;
```

Comment:
I noticed that you removed image input support. I don't think we should do that. It's useful to be able to run the demo on individual images.

Reply:
`cv::VideoCapture` can process single image as input. New implementation lacks only ability to read from a folder (BTW it seems `VideoCapture` can't read from folders because it is unclear how to do it: what the order of images should be, should it be recursive. And that pattern `%04d.jpg` for images was introduced instead). Supporting both ways will make the code harder to read because there will be `if` deciding how to iterate over the input. The solution could be to introduce a wrapper handling different types of inputs, maybe even implementing iterators for it. But until then I think it is better not to add such functionality, especially taking into account that most demos also don't have that feature.

Reply:
I don't think we need folder support (the ability to use patterns should make up for that).

However, single image support is useful, and I don't think the current implementation supports it well enough. I just tried it and when you give the demo an image it quits immediately, before the output window even has time to appear. I think we should at least start the demo in a paused state if we detect that a single image was given.

Reply:
I agree. A check, that VideoCapture has only one image will be not clear because there is no such method answering exactly that question, and we can't just always start the demo paused because of tests. This pausing can be regulated by `-no_show` or delay key could be introduced. Delay key would also enable kind of slide show with a number like 1000. And it will be clearer that pausing depends on delay but not `-no_show`. But I don't know whether such functionality with delay key is needed.

Reply:
Updated

Reply:
OK, I think this is a reasonable solution. Although, FWIW:

> A check, that VideoCapture has only one image will be not clear because there is no such method answering exactly that question

Wouldn't comparing `CAP_PROP_FRAME_COUNT` against 1 answer it?

Reply:
A video with a single frame will also has `CAP_PROP_FRAME_COUNT ` 1. We could pretend we don't care whether it is a short video or an image, but a video of 2 frames should probably also run paused. It becomes unclear how to decide at which value to run the demo paused.

Reply:
OK, fair enough.